cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -809,21 +801,45 @@ func (c *client) kubectlManifestCmd(commandName, manifest string) error {
 	return nil
 }
 
-func (c *client) buildKubectlArgs(commandName string) []string {
+func (c *client) buildKubectlManifestArgs(commandName string) []string {
 	args := []string{commandName}
+	if c.configOverrides.Context.Namespace != "" {
+		args = append(args, "--namespace", c.configOverrides.Context.Namespace)
+	}
+	args = append(args, c.buildKubectlArgs()...)
+	return append(args, "-f", "-")
+}
+
+func (c *client) kubectlDeleteAllCmd(namespace, resource string) error {
+	cmd := exec.Command("kubectl", c.buildKubectlDeleteAllArgs(namespace, resource)...) //nolint:gosec
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "couldn't kubectl delete %s --all, output: %s", resource, string(out))
+	}
+	return nil
+}
+
+func (c *client) buildKubectlDeleteAllArgs(namespace, resource string) []string {
+	args := []string{"delete", resource}
+	if namespace != "" {
+		args = append(args, "--namespace", namespace)
+	}
```

Comment:
If you want to use kubectl to delete from all namespaces, you'll need to specify `--all-namespaces`; otherwise, it will only delete from the namespace for the kubeconfig's context.

Reply:
good catch @ncdc